core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -564,6 +564,13 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
     val encryptionEnabled = get(NETWORK_ENCRYPTION_ENABLED) || get(SASL_ENCRYPTION_ENABLED)
     require(!encryptionEnabled || get(NETWORK_AUTH_ENABLED),
       s"${NETWORK_AUTH_ENABLED.key} must be enabled when enabling encryption.")
+
+    val executorTimeoutThreshold = Utils.timeStringAsSeconds(get("spark.network.timeout", "120s"))
+    val executorHeartbeatInterval = Utils.timeStringAsSeconds(
+      get("spark.executor.heartbeatInterval", "10s"))
+    require(executorHeartbeatInterval > executorTimeoutThreshold, s"The value of" +
```

Comment:
Should also add a simple and explicit comment to explain the reason why this check is needed.

Reply:
Done